handler/src/main/java/io/netty/handler/ipfilter/RuleBasedIpFilter.java

```diff
@@ -22,36 +22,71 @@
 
 import java.net.InetSocketAddress;
 import java.net.SocketAddress;
+import java.util.ArrayList;
+import java.util.List;
 
 /**
+ * <p>
  * This class allows one to filter new {@link Channel}s based on the
  * {@link IpFilterRule}s passed to its constructor. If no rules are provided, all connections
  * will be accepted.
+ * </p>
  *
+ * <p>
  * If you would like to explicitly take action on rejected {@link Channel}s, you should override
  * {@link AbstractRemoteAddressFilter#channelRejected(ChannelHandlerContext, SocketAddress)}.
+ * </p>
+ *
+ * <p> Consider using {@link IpSubnetFilter} for better performance while not as
+ * general purpose as this filter. </p>
  */
 @Sharable
 public class RuleBasedIpFilter extends AbstractRemoteAddressFilter<InetSocketAddress> {
 
-    private final IpFilterRule[] rules;
+    private final boolean acceptIfNotFound;
+    private final List<IpFilterRule> rules;
 
+    /**
+     * <p> Create new Instance of {@link RuleBasedIpFilter} and filter incoming connections
+     * based on their IP address and {@code rules} applied. </p>
+     *
+     * <p> {@code acceptIfNotFound} is set to {@code true}. </p>
+     *
+     * @param rules An array of {@link IpFilterRule} containing all rules.
+     */
     public RuleBasedIpFilter(IpFilterRule... rules) {
-        this.rules = ObjectUtil.checkNotNull(rules, "rules");
+        this(true, rules);
     }
 
-    @Override
-    protected boolean accept(ChannelHandlerContext ctx, InetSocketAddress remoteAddress) throws Exception {
+    /**
+     * Create new Instance of {@link RuleBasedIpFilter} and filter incoming connections
+     * based on their IP address and {@code rules} applied.
+     *
+     * @param acceptIfNotFound If {@code true} then accept connection from IP Address if it
+     *                         doesn't match any rule.
+     * @param rules            An array of {@link IpFilterRule} containing all rules.
+     */
+    public RuleBasedIpFilter(boolean acceptIfNotFound, IpFilterRule... rules) {
+        ObjectUtil.checkNotNull(rules, "rules");
+
+        this.acceptIfNotFound = acceptIfNotFound;
+        this.rules = new ArrayList<IpFilterRule>();
```

Comment:
nit: Please do use `new ArrayList(rules.length)`

Reply:
If the `rules` array contains `null`, we will waste Arraylist capacity because we filter `null` rules later on.

Reply:
its very unlikely it contains null... so better use the length so we not need to "grow" the list.

Reply:
Done